plugins/carlabase/CMakeLists.txt

```diff
@@ -6,7 +6,29 @@ IF(NOT CMAKE_VERSION VERSION_LESS 3.9)
   CMAKE_POLICY(SET CMP0068 OLD)
 ENDIF()
 
-if(LMMS_HAVE_CARLA)
+# If Carla was not provided by the system, make a dummy library instead
+if(LMMS_HAVE_WEAKCARLA)
+  # Mimic the Makefile header
+  FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/config.h "")
+  SET(CARLA_INCLUDE_DIRS
+    ${CMAKE_CURRENT_SOURCE_DIR}/carla/source
+    ${CMAKE_CURRENT_SOURCE_DIR}/carla/source/includes
+    ${CMAKE_CURRENT_SOURCE_DIR}/carla/source/utils
+    ${CMAKE_CURRENT_SOURCE_DIR}/carla/source/backend
+    ${CMAKE_CURRENT_BINARY_DIR}
```

Comment:
@tresf This line will break MinGW builds once merged into `stable-1.2` because it affects the search order of `plugin_export.h`. Can we use some other path for the dummy `config.h`?

Reply:
> once merged into stable-1.2

It is merged... `tresf merged 13 commits into stable-1.2 from xenial 17 days ago`

> because it affects the search order of `plugin_export.h`

Can you help me understand where `plugin_export.h` is created?  I'm just not sure the exact issue.

> Can we use some other path for the dummy config.h?

So I assume then the problem is this line... 

```cmake
INCLUDE_DIRECTORIES(${CARLA_INCLUDE_DIRS})
```

Ideally, [`TARGET_INCLUDE_DIRECTORIES`] should really be favored but the build is farmed off to a plugin, so I wasn't aware of a standard way to do this.

I'm still a bit confused though.  Is the issue that another plugin that Mingw uses has a file also called `config.h`?  What build system is failing, is it `master`, not `stable-1.2`?

To answer the question, the file can be written anywhere, but I'm a bit confused why it would actually help.  I'd expect the `carlabase` build (i.e. `CMAKE_CURRENT_BINARY_DIR`) directory to be about as sane as it gets.

Reply:
Sorry, I mean `master`.

Reply:
@PhysSong if it's a search order problem, I'd expect it to be fixed by changing this... 

```cmake
- INCLUDE_DIRECTORIES(${CARLA_INCLUDE_DIRS})
+ INCLUDE_DIRECTORIES(AFTER ${CARLA_INCLUDE_DIRS})
```

... but cmake claims that the default behavior is already `AFTER`.  I guess I just don't understand how this is happening... Is it that the newer version of mingw is writing the same file in the same place?  If that's the case, can we just wrap a check around it?

Reply:
I think `${CMAKE_CURRENT_BINARY_DIR}` is causing the problem. If we use something like `${CMAKE_CURRENT_BINARY_DIR}/carla`, that shouldn't interfere with the search order.

Reply:
Done via a77e592.